app/src/main/java/com/github/libretube/helpers/PlayerHelper.kt

```diff
@@ -532,22 +527,40 @@ object PlayerHelper {
             }
     }
 
+
+    /**
+     * Returns the recommended duration that should be buffer in milliseconds, based on the available memory.
+     */
+    private fun getRecommendedBufferMs(): Int {
+        val activityManager =
+            LibreTubeApp.instance.getSystemService(Context.ACTIVITY_SERVICE) as ActivityManager
+        val memInfo = ActivityManager.MemoryInfo().also { memoryInfo ->
+            activityManager.getMemoryInfo(memoryInfo)
+        }
+        // available memory in MB
+        val availableMemory = memInfo.availMem / (1024 * 1024)
+        return when {
+            // always use min buffer when on low memory
+            (memInfo.lowMemory) -> MINIMUM_BUFFER_DURATION
+            availableMemory >= 8192 -> 1000 * 450
+            availableMemory >= 4096 -> 1000 * 300
+            availableMemory > 2048 -> 1000 * 60 * 2
+            else -> MINIMUM_BUFFER_DURATION
+        }
+    }
+
     /**
      * Get the load controls for the player (buffering, etc)
      */
     @OptIn(androidx.media3.common.util.UnstableApi::class)
-    fun getLoadControl(): LoadControl {
-        return DefaultLoadControl.Builder()
-            // cache the last three minutes
-            .setBackBuffer(1000 * 60 * 3, true)
-            .setBufferDurationsMs(
-                MINIMUM_BUFFER_DURATION,
-                max(bufferingGoal, MINIMUM_BUFFER_DURATION),
-                DefaultLoadControl.DEFAULT_BUFFER_FOR_PLAYBACK_MS,
-                DefaultLoadControl.DEFAULT_BUFFER_FOR_PLAYBACK_AFTER_REBUFFER_MS
-            )
-            .build()
-    }
+    fun getLoadControl(): LoadControl = DefaultLoadControl.Builder()
```

Comment:
I think this method is only called when the player is created for the first time, so it stays the same during autoplay or starting other videos (except if the current player is getting closed by the user).

But isn't it very likely that the available RAM is going to change when playing the next video, e.g. via autoplay?

Reply:
The method also isn't very accurate, since it doesn't include memory used by the kernel :) 
Changed it to be based on the total memory that the device has.

Alternatively we could also just choose a value, but in either case I would like to remove this setting, as it's not really user-friendly or helpful.

Reply:
The PR actually tries to simplify the preferences and remove "unnecessary"/"unintuitive" preferences that are not really easy to understand or useful.

So adding a new option to toggle between auto and custom would be like the opposite of what this PR is trying to achieve imho.

Reply:
> Sorry for comments here but as a user I feel there should be two options in setting " auto" and "custom" . You can set "auto" by default but if some users want to change it they will.

What's the issue with the (current) default, that requires you changing it?

> For me this setting save a lots of data uses

The (network) data that is consumed is the same, it's just distributed differently.